### PR TITLE
Feature/services

### PR DESCRIPTION
**Pull request by MarkB-Mugoh**
**Hello All**
The purpose of this pull request is to kindly request your valuable review of the ServicesService class implementation. I invite you to provide your insights, suggestions, and identify areas where improvements can be made to enhance the design, performance, or overall effectiveness of the service logic.

**Key Highlights:**

> ServicesService Class

-  class has been structured to facilitate extensibility for specific service functionalities.
- Routes and handlers for the Contact and Book services have been defined, ensuring seamless integration with the core logic.
- Placeholder logic has been included for handling contact submissions and device bookings.
- The class has been documented with JavaDoc comments to enhance code readability and maintainability.

### **Note:**
I do not know the logic to process these services, so that part is left out untill @otkothtuk can explain the logic of fetching the data from the IOT Devices

### CODE REVIEW

**Comment by WesleyJamesObuya** [comment]
Noted

File: src/main/java/com/seed/api/core/ServicesService.java [hunk 1]
```diff
@@ -17,13 +30,152 @@ public class ServicesService extends TransactionsService {
...
 
     /**
-     * Sets routes for the http server.
+     * Sets routes for the http server, including
+     * routes for additional services.
+     *
      * @param router The router used to set paths.
      */
     protected void setServicesRoutes(final Router router) {
+        // Set the routes for location.
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/ServicesService.java:39]: router.post("/contact/submit").handler(this::handleContactSubmission);
router.post("/slot/book").handler(this::handleSlotBooking);
router.post("/book/device").handler(this::handleDeviceBooking);

File: src/main/java/com/seed/api/core/ServicesService.java [hunk 2]
```diff
@@ -17,13 +30,152 @@ public class ServicesService extends TransactionsService {
...
+    |
+    |
+    | 1. CONTACT US SERVICE
+    |
+    |
+    */
+    protected void setContactRoutes(final Router router) {
+        router.route("/contact/*").handler(BodyHandler.create());
+
+        CorsHandler corsHandler = CorsHandler.create("*")
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/ServicesService.java:62]: This is not necessary. Its been done at the CoreService.java class

File: src/main/java/com/seed/api/core/ServicesService.java [hunk 3]
```diff
@@ -17,13 +30,152 @@ public class ServicesService extends TransactionsService {
...
+            .allowedMethods(Set.of(HttpMethod.POST));
+
+        router.route("/contact/*").handler(corsHandler);
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/ServicesService.java:65]: Not necessary

File: src/main/java/com/seed/api/core/ServicesService.java [hunk 4]
```diff
@@ -17,13 +30,152 @@ public class ServicesService extends TransactionsService {
...
+     */
+
+    /*
+    |
+    |
+    | 2. Book Slot
+    |
+    |
+    */
+    protected void setSlotRoutes(final Router router) {
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/ServicesService.java:105]: Not necessary.

File: src/main/java/com/seed/api/core/ServicesService.java [hunk 5]
```diff
@@ -17,13 +30,152 @@ public class ServicesService extends TransactionsService {
...
+     * @param router The router used to set paths.
+     */
+    /*
+    |
+    |
+    | 1. CONTACT US SERVICE
+    |
+    |
+    */
+    protected void setContactRoutes(final Router router) {
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/ServicesService.java:59]: Not necessary.

File: src/main/java/com/seed/api/core/ServicesService.java [hunk 6]
```diff
@@ -17,13 +30,152 @@ public class ServicesService extends TransactionsService {
...
+     * @param router The router used to set paths.
+     */
+    /*
+    |
+    |
+    | 3. Book to collect Device
+    |
+    |
+    */
+    protected void setBookRoutes(final Router router) {
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/ServicesService.java:150]: Not necessary.

File: src/main/java/com/seed/api/core/ServicesService.java [hunk 7]
```diff
@@ -17,13 +30,152 @@ public class ServicesService extends TransactionsService {
...
+     * routes for additional services.
+     *
      * @param router The router used to set paths.
      */
     protected void setServicesRoutes(final Router router) {
+        // Set the routes for location.
+        this.setTransactionsRoutes(router);
+
+        // Add routes for additional services (Contact and Book)
+        setContactRoutes(router);
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/ServicesService.java:43]: Not necessary

File: src/main/java/com/seed/api/core/ServicesService.java [hunk 8]
```diff
@@ -17,13 +30,152 @@ public class ServicesService extends TransactionsService {
...
+        setBookRoutes(router);
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/ServicesService.java:44]: Not necessary.

File: src/main/java/com/seed/api/core/UserService.java [hunk 9]
```diff
@@ -1,29 +1,136 @@
...
     /**
-     * Sets routes for the http server.
+     * Sets routes for the http server related
+     * to user management (login and registration).
      * @param router The router used to set paths.
      */
+
     protected void setUserRoutes(final Router router) {
+        // Enable parsing of request body for POST requests
+        router.route().handler(BodyHandler.create());
```
Comment (otkothtuk) [inline src/main/java/com/seed/api/core/UserService.java:33]: Not necessary

**Review by otkothtuk — APPROVED** [review]

**Pull request merged**